src/components/Common/DialogWithReduxState.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import { Provider, ReactReduxContext } from 'react-redux'
+import Dialog from 'material-ui/Dialog'
+
+export default function DialogWithReduxState(props) {
```

Comment:
The change from the mui Dialog to the `DialogWithReduxState` isn't clear to me. So I see that the new `DialogWithReduxState` has a provider that it injects the redux store into. But in looking at `AssignToDialog.js` for instance, which uses `DialogWithReduxState`, I don't see `AssignToDialog` using the supplied store directly. Am I missing something?

Reply:
The store is not being used by `AssignToDialog` but one of its children, `ToolbarExtendedChoicePicker`.

Als for some more general info on why this is needed see [this Slack thread](https://dhis2.slack.com/archives/CPGUFVC56/p1603716699281200).

Reply:
Ok. I see a suggestionfield mentioned in that discussion which I can't find in the ToolbarExtendedChoicePicker, so I'm assuming that the discussion is about a similar area of code, where a redux provider wasn't accessible through a portal?

I see how providing a provider in the component that creates the portal solves the issue, but it does not make it clear to me as someone unfamiliar with the codebase which component down the tree is consuming the store. Would it be possible to use the store provider further down the tree, closer to the component that needs it? That would also optimize the rerendering.

Reply:
> I see how providing a provider in the component that creates the portal solves the issue, but it does not make it clear to me as someone unfamiliar with the codebase which component down the tree is consuming the store.

I'm not really following this line of reasoning.... The way I see it, the clearest solution is one that is in close proximity of the problem. The problem in this case is that the context is not passed onto a portal, so IMO the cleanest solution is to fix it in the component that renders in a portal. And continuing this line of reasoning, the other part of your argument about "_which component down the tree is consuming the store_" also doesn't seem that relevant: in a redux app we should assume that any component in the hierarchy is able to tap into the app-state. This was not possible initially, but after introducing `DialogWithReduxState` it was.

> That would also optimize the rerendering.

IMO that would be a valid argument only if the rendering needs to be optimised

Reply:
Ah wait sorry, I see the miscommunication now, my bad. I was conflating connect with the provider. So if I understand correctly the portal basically needs its own redux provider just like the app root has. So yes, in that case your solution makes sense to me and is what I'd expect as well.

I'd add an explanatory comment to the dialog that the redux provider doesn't work across portals, just so that devs are aware what's going on there. It's news to me that context doesn't work across portals, might be to others as well.